scripts/apishow.py

```diff
@@ -1,55 +1,235 @@
+            collection_len = len(collection)
+        except AttributeError:
+            collection_len = 0
+
+        if collection_len > 0:
+            _init_store('entity', store[collection.name])
+            entity = random.choice(collection)
+
+            try:
+                actions = entity.action.all
+            except KeyError:
+                actions = []
+            if 'actions_avail' not in store[collection.name]['entity']:
+                store[collection.name]['entity']['actions_avail'] = set(actions)
+
+            if collection.name in subcollections:
+                for subcol_name in subcollections[collection.name]:
+                    try:
+                        subcol = getattr(entity, subcol_name)
+                    except AttributeError:
+                        continue
+                    _process_collection(subcol, store[collection.name])
+
+    for collection in api.collections.all:
+        _process_collection(collection, store)
+
+
+def print_info(store):
+    """Print info about collections together with coverage info when available."""
+
+    for name, collection in sorted(store.iteritems()):
+        print('=' * (2 + len(name)))
+        print('* {}'.format(name))
+
+        def _print_resource(res_title, res_dict):
+            if 'actions_avail' in res_dict and res_dict['actions_avail']:
+                print('  {} actions:'.format(res_title))
+                covered = True if 'actions' in res_dict else False
+                for action in res_dict['actions_avail']:
+                    print('    * {}{}'.format(
+                        action,
+                        ' OK' if covered and action in res_dict['actions'] else ''))
+            if 'entity' in res_dict:
+                _print_resource('{} entity'.format(res_title), res_dict['entity'])
+
+            for key, subcollection in sorted(res_dict.iteritems()):
+                if key in ('actions', 'actions_avail', 'entity', 'methods'):
+                    continue
+                _print_resource('Subcollection "{}"'.format(key), subcollection)
+
+        _print_resource('Collection', collection)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument(
+        '--address',
+        default=conf.env.get('base_url', None),
+        help="hostname or ip address of target appliance")
+    parser.add_argument(
+        '--logfile',
+        metavar='FILE',
+        default=None,
+        help="path to cfme log file")
+    args = parser.parse_args()
+
+    # we are really not interested in any warnings and "warnings.simplefilter('ignore')"
+    # doesn't work when it's redefined later in the REST API client
+    warnings.showwarning = lambda *args, **kwargs: None
+
+    api = MiqApi(
+        '{}/api'.format(
+            args.address if not args.address.endswith('/') else re.sub(r'/+$', '', args.address)),
```

Comment:
**optional**
Why not just use args.address.rstrip('/') here? It will strip all trailing slashes or just return the address as-is without using re.sub() or a ternary statement